From 808fdc26fe6a409dd686df4ae1ce1faf238bead0 Mon Sep 17 00:00:00 2001
From: Toni Barzic <tbarzic@chromium.org>
Date: Thu, 07 Nov 2024 00:50:12 +0000
Subject: [PATCH] [Merge to 130] Don't show history clusters menu item if history clusters is disabled.

Prior to this CL, history clusters could be disabled but Chrome would
continue to show the history clusters menu item.

Bug: 374013418, 376751203
Change-Id: Ia606dd4cb4fb6c1c1b8a284839ef6484fe6fdc83
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5991910
Reviewed-by: Erik Chen <erikchen@chromium.org>
Commit-Queue: Toni Barzic <tbarzic@google.com>
Cr-Commit-Position: refs/branch-heads/6723@{#1840}
Cr-Branched-From: 985f2961df230630f9cbd75bd6fe463009855a11-refs/heads/main@{#1356013}
---

--- a/chrome/browser/ui/tabs/recent_tabs_sub_menu_model.cc
+++ b/chrome/browser/ui/tabs/recent_tabs_sub_menu_model.cc
@@ -34,10 +34,12 @@
 #include "chrome/browser/ui/browser_commands.h"
 #include "chrome/browser/ui/browser_live_tab_context.h"
 #include "chrome/browser/ui/browser_window.h"
+#include "chrome/browser/ui/browser_window/public/browser_window_features.h"
 #include "chrome/browser/ui/tabs/tab_group_theme.h"
 #include "chrome/browser/ui/tabs/tab_strip_model.h"
 #include "chrome/browser/ui/toolbar/app_menu_model.h"
 #include "chrome/browser/ui/ui_features.h"
+#include "chrome/browser/ui/views/side_panel/history_clusters/history_clusters_side_panel_coordinator.h"
 #include "chrome/grit/generated_resources.h"
 #include "components/favicon/core/history_ui_favicon_request_handler.h"
 #include "components/favicon_base/favicon_types.h"
@@ -240,6 +242,10 @@
   if (!custom_commands.contains(command_id)) {
     return false;
   }
+  if (command_id == IDC_SHOW_HISTORY_CLUSTERS_SIDE_PANEL &&
+      !HistoryClustersSidePanelCoordinator::IsSupported(browser_->profile())) {
+    return false;
+  }
   if (log_menu_metrics_callback_) {
     log_menu_metrics_callback_.Run(command_id);
   }
@@ -337,11 +343,12 @@
   InsertItemWithStringIdAt(0, IDC_SHOW_HISTORY, IDS_HISTORY_SHOW_HISTORY);
   SetCommandIcon(this, IDC_SHOW_HISTORY,
                  vector_icons::kHistoryChromeRefreshIcon);
-
-  InsertItemWithStringIdAt(1, IDC_SHOW_HISTORY_CLUSTERS_SIDE_PANEL,
-                           IDS_HISTORY_CLUSTERS_SHOW_SIDE_PANEL);
-  SetCommandIcon(this, IDC_SHOW_HISTORY_CLUSTERS_SIDE_PANEL,
-                 vector_icons::kHistoryChromeRefreshIcon);
+  if (HistoryClustersSidePanelCoordinator::IsSupported(browser_->profile())) {
+    InsertItemWithStringIdAt(1, IDC_SHOW_HISTORY_CLUSTERS_SIDE_PANEL,
+                             IDS_HISTORY_CLUSTERS_SHOW_SIDE_PANEL);
+    SetCommandIcon(this, IDC_SHOW_HISTORY_CLUSTERS_SIDE_PANEL,
+                   vector_icons::kHistoryChromeRefreshIcon);
+  }
 
   AddSeparator(ui::NORMAL_SEPARATOR);
   history_separator_index_ = GetItemCount() - 1;
--- a/chrome/browser/ui/tabs/recent_tabs_sub_menu_model_unittest.cc
+++ b/chrome/browser/ui/tabs/recent_tabs_sub_menu_model_unittest.cc
@@ -20,6 +20,7 @@
 #include "base/test/bind.h"
 #include "build/build_config.h"
 #include "chrome/app/chrome_command_ids.h"
+#include "chrome/browser/history/history_service_factory.h"
 #include "chrome/browser/sessions/chrome_tab_restore_service_client.h"
 #include "chrome/browser/sessions/exit_type_service.h"
 #include "chrome/browser/sessions/session_service.h"
@@ -142,6 +143,12 @@
     EnableSync();
   }
 
+  TestingProfile::TestingFactories GetTestingFactories() override {
+    return {TestingProfile::TestingFactory{
+        HistoryServiceFactory::GetInstance(),
+        HistoryServiceFactory::GetDefaultFactory()}};
+  }
+
   void EnableSync() {
     // ClientTagBasedDataTypeProcessor requires connecting before
     // other interactions with the worker happen.
